include/z64cutscene_commands.h

```diff
@@ -11,445 +11,274 @@
- *   size = 0x30
+ * Fade out the sequence that is playing on the specified sequence player, over the specified frame range.
+ * @see `CutsceneFadeOutSeqPlayer`
  */
-#define CS_FADE_BGM(fadeType, startFrame, endFrame, unused0, unused1, unused2, unused3, unused4, unused5, unused6, unused7) \
-    CMD_HH(fadeType, startFrame), CMD_HH(endFrame, unused0), \
+#define CS_FADE_OUT_SEQ(seqPlayer, startFrame, endFrame, unused0, unused1, unused2, unused3, unused4, unused5, unused6, unused7) \
+    CMD_HH(seqPlayer, startFrame), CMD_HH(endFrame, unused0), \
     CMD_W(unused1), CMD_W(unused2), CMD_W(unused3), CMD_W(unused4), CMD_W(unused5), \
     CMD_W(unused6), CMD_W(unused7), 0x00000000, 0x00000000, 0x00000000
 
 /**
- * ARGS
- *   s32 entries (e)
- * FORMAT
- *   0000008C eeeeeeee
- *   size = 0x8
+ * Declares a list of `CS_TIME` entries.
  */
-#define CS_TIME_LIST(entries) CS_CMD_SETTIME, CMD_W(entries)
+#define CS_TIME_LIST(entries) \
+    CS_CMD_TIME, CMD_W(entries)
 
 /**
- * ARGS
- *   s16 unk (u), s16 startFrame (s), s16 endFrame (e), s8 hour (h), s8 min (m)
- * FORMAT
- *   Capital U is Unused
- *   uuuussss eeeehhmm UUUUUUUU
- *   size = 0xC
+ * Sets the time of day.
+ * Both the day time and skybox time are set by this command.
  */
-#define CS_TIME(unk, startFrame, endFrame, hour, min, unused) \
-    CMD_HH(unk, startFrame), \
-    CMD_HBB(endFrame, hour, min), \
-    CMD_W(unused)
+#define CS_TIME(unused0, frame, unused1, hour, min) \
+    CMD_HH(unused0, frame), \
+    CMD_HBB(unused1, hour, min), \
+    CMD_W(0)
 
 /**
- * ARGS
- *   CutsceneTerminatorDestination dest (d), s16 startFrame (s), s16 endFrame (e)
- * FORMAT
- *   Capital U is Unused , endFrame duplicate
- *   000003E8 00000001 ddddssss eeeeUUUU
- *   size = 0x10
+ * Sends the player to a new destination. 
+ * `destination` maps to a custom block of code that must implement the scene transition on its own.
+ * This custom code can also do other tasks like changing age, setting flags, or any other setup that is needed
+ * before going to the next destination.
+ * 
+ * @see `CutsceneDestination`
  */
-#define CS_TERMINATOR(dest, startFrame, endFrame) \
-    CS_CMD_TERMINATOR, 0x00000001, CMD_HH(dest, startFrame), CMD_HH(endFrame, endFrame)
+#define CS_DESTINATION(destination, frame, unused) \
+    CS_CMD_DESTINATION, 1, CMD_HH(destination, frame), CMD_HH(unused, unused)
 
 /**
- * Marks the end of a cutscene
+ * Marks the end of a cutscene script.
  */
 #define CS_END() 0xFFFFFFFF, 0x00000000
 
+
+// most instances of this look like unimplemented actor cues.
+// after confirming this, can probably switch those over and delete this
+#define CS_UNK_DATA_LIST(cmdType, entries) CMD_W(cmdType), CMD_W(entries)
+
+#define CS_UNK_DATA(unk1, unk2, unk3, unk4, unk5, unk6, unk7, unk8, unk9, unk10, unk11, unk12) \
+    CMD_W(unk1), CMD_W(unk2), CMD_W(unk3), CMD_W(unk4), CMD_W(unk5), CMD_W(unk6), \
+    CMD_W(unk7), CMD_W(unk8), CMD_W(unk9), CMD_W(unk10), CMD_W(unk11), CMD_W(unk12)
```

Comment:
Looked through these, and all of them look like valid actor cues.

Reply:
Indeed, as my comment notes. 
I went through all of them and have an enum ready for them to use when ZAP is updated in the future